docs/DeveloperGuide.adoc

```diff
@@ -316,39 +319,141 @@ _{More to be added}_
 (For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Create meme
+
+*MSS*
+
+1. User requests to list meme templates
+2. Weme shows a list of meme template
+3. User selects a template to create a meme
+4. Weme creates the meme from the selected template based on user input
++
+Use case ends
+
+*Extensions*
+
+[none]
+* 2a. The template list is empty
+[none]
+** 2a1. User adds template
++
+2a2. Weme interactively generates template based on user input
++
+Use case ends
+
+* 3a. The given index is invalid
++
+[none]
+** 3a1. Weme shows an error message
++
+Use case resumes at step 2
+
+* 4a. The given input is invalid
+
+
+[discrete]
+=== Use case: Create template
+
+*MSS*
+
+1. User opens the template tab
+2. Weme opens the template tab
+3. User requests to create a template, using a base picture from the disk
+4. Weme displays the base picture in the editing area
+5. User requests to add/move text boxes in the base picture
+6. Weme adds/moves text boxes in the base picture
+7. User completes template creation
+8. Weme successfully creates the new template
++
+Use case ends
+
+*Extensions*
+[none]
+* 3. The given path is invalid
+[none]
+** 3a1. AddressBook shows an error message
++
+Use case resumes at step 2
+
+[discrete]
+=== Use case: Import meme
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User toggles the import tab
+2. Weme opens the import tab
+3. User requests to import a meme collection from a directory
+4. Weme adds the memes from the collection into staging area
+5. User deletes unwanted memes in the staging area
+6. Weme deletes those memes from the staging area
+7. User confirms the import
+8. Weme successfully imports the selected memes from the collection
 +
-Use case ends.
+Use case ends
 
 *Extensions*
+[none]
+* 3. The given directory path is invalid.
+[none]
+** 3a1. AddressBook shows an error message
++
+Use case resumes at step 2
 
+* 3. The given collection is corrupted
++
 [none]
-* 2a. The list is empty.
+** 3a1. Weme shows an error message
 +
-Use case ends.
+Use case resumes at step 2
+
+
+[discrete]
+=== Use case: Undo/Redo
+
+*MSS*
 
-* 3a. The given index is invalid.
+1. User opens Weme
+2. User requests to delete a template
+3. Weme deletes the template
+4. User requests to undo
+5. Weme undoes the deletion restores the template
+6. User requests to redo
+7. Weme redoes and re-executes step 3
+8. User logs out
 +
+Use case ends
+
+
+*Extensions*
+[none]
+* 5. User keys in undo again
+[none]
+** 5a1. AddressBook shows an error message
++
+Use case resumes at step 1
+
+* 5. User adds a template and keys in redo
 [none]
-** 3a1. AddressBook shows an error message.
+** 5a1. AddressBook shows an error message
```

Comment:
Please do a search and replace to change all occurrances of `AddressBook` to `Weme`?  Maybe limit the range to only the Appendix.